### PR TITLE
(1) Remove OASIS XSL StyleSheets from GnuCash-docs Tree

### PR DESCRIPTION
**Pull request by usamimikamen-sk**
In this PR, OASYS XSL stylesheets from GnuCash-docs are removed completely.
1. 522508d
Added an argument dtd_files to add_${fmt}_target and added dependency for DTD files. 
Changed build directory creation timing from configuration to build.
Prepared variables for build directory transition.

Issue Fixed:
* Dependency for language-specific DTD files is missing.
* Create dependency for GnuCash-customized xsl files.

2. fcf64b3
Merged MakeChm.cmake into AddChmTarget.cmake.
Because functions of context-specific help for CHM are not used in GnuCash, the map section and alias section are removed.

3. d1134db
Added missing OASIS xslt icons into xsl/images.
Put latest GnuCash-specific icons into xsl/icons.
Removed stylesheets/.
Changed CMake files related the new directory structure.
Changed ${BUILD_DIR} to ${CMAKE_CURRENT_BINARY_DIR}/${fmt} for CHM, PDF, and EPUB target.

${BUILD_DIR} is changed in this commit in order to add target clean-extra in the future.
Additionally, the directory for the target ghelp is changed to share/help/${lang}/${docname}.

The location of output files may be changed here but will be restored later.

Issue Fixed:
* Move contents of stylesheet/ to xsl/.
* GnuCash-specific icons are wanted to be used.
* Create dependency for GnuCash-specific icons.
* Some admonition and callout icons are lacking.
* Admonition icons directory is incorrect for chm.

4. d455af1 and 54962ae
Added new XSL structure without customization.
Applied new XSL files for each format.
Removed xsl/1.79.2/ completely.
Removed unused customization files.

Current customizations are removed but will be restored later

Issue Fixed:
* Remove OASIS XSLT from xsl/1.79.2 and use system xsl files.

5. fe36eda and 9dbed25
Changed the layout of ${OUTPUT_DIR}.
Fix target install for each format.
Clean up unused variables.

The output location of PDF, EPUB, MOBI, and CHM is restored.

Issue Fixed:
* Lacking dependency: ${lang}-${docname}-mobi ${lang}-${docname}-epub.
 
6. 72ce9e2 and 14dd610
Restored customizations that don't have side effects.
Apply fixed titlepage and variablelist.

FO and HTML have different formats, so you need to customize them appropriately depending on how docbook-xsl is written.
Rewrote titlepage and variablelist for each format.

Issue Fixed:
* Current variablelist.xsl breaks docbook-xsl multiple features, should be split for HTML and FO.
 * Customization of titlepage is not applied, should be split for HTML and FO.
 * Resolution of PNG admonition icons is too low. Use SVG instead.
 * Customize variable lists, PR #203.

7. 046e895
Change paper size setting:
US letter for C,
A4 for other languages.

ISO A4 paper size is used worldwide but letter and legal size
are mainly used in the US. Therefore paper size is set above.
This setting should be moved CMake option in the future.

Issue Fixed:
* Paper size for PDF is set USletter (except LANG C.)

8. f1945c8
Set character code Shift_JIS and KOI8-R for Japanese and Russian, respectively.

CHM format uses traditional Windows codepages, not Unicode. Therefore
traditional character code should be applied for the HTML and TOC.

Issue Fixed:
* TOC of chm for ja and ru is not shown correctly.

### CODE REVIEW

**Comment by fellen** [comment]
About  commit d1134db: What is wrong with the [default images](https://github.com/docbook/xslt10-stylesheets/tree/master/xsl/images)?

**Comment by usamimikamen-sk** [comment]
Originally, xsl/1.79.2/images, stylesheets/* except for gnucash-icon.*,  and https://github.com/docbook/xslt10-stylesheets/tree/master/xsl/images are the same. But PNG files are compressed by https://github.com/Gnucash/gnucash-docs/pull/124.
So I selected these files.

**Comment by fellen** [comment]
> Originally, xsl/1.79.2/images, stylesheets/* except for gnucash-icon.*, and https://github.com/docbook/xslt10-stylesheets/tree/master/xsl/images are the same. But PNG files were compressed by #124. So I selected these files.

Perhaps you would better create an optiping PR for https://github.com/docbook/xslt10-stylesheets?
That  would save us the maintenance burden and others would benefit, too.

File: xsl/gnc-custom-chm.xsl [hunk 1]
```diff
@@ -20,6 +20,26 @@
...
 
+<!-- language specific character code for traditional Windows codepages
+     ja: Shift_JIS
+     ru: KOI8-R
+-->
+<xsl:param name="htmlhelp.encoding">
+  <xsl:choose>
+    <xsl:when test="$gnc.lang = 'ja'">Shift_JIS</xsl:when>
+    <xsl:when test="$gnc.lang = 'ru'">KOI8-R</xsl:when>
+    <xsl:otherwise>ISO-8859-1</xsl:otherwise>
```
Comment (fellen) [inline xsl/gnc-custom-chm.xsl:31]: If UTF-8 is really no option we need ISO 8859-15 for Americas, Western Europe, Oceania, and much of Africa
Otherwise we have no currency symbol.
[ISO/IEC 8859-15](https://en.wikipedia.org/wiki/ISO/IEC_8859-15)

File: xsl/gnc-custom-chm.xsl [hunk 2]
```diff
@@ -20,6 +20,26 @@
...
+  </xsl:choose>
+</xsl:param>
+
+<xsl:param name="chunker.output.encoding">
+  <xsl:choose>
+    <xsl:when test="$gnc.lang = 'ja'">Shift_JIS</xsl:when>
+    <xsl:when test="$gnc.lang = 'ru'">KOI8-R</xsl:when>
+    <xsl:otherwise>ISO-8859-1</xsl:otherwise>
```
Comment (fellen) [inline xsl/gnc-custom-chm.xsl:39]: here too

**Review by fellen — CHANGES_REQUESTED** [review]

File: xsl/gnc-custom-chm.xsl (hunk 1 above)
Comment (usamimikamen-sk) [reply xsl/gnc-custom-chm.xsl:31]: Windows code pages are not compliant with ISO codes like ISO-8859-*. Moreover, the multibyte charset is more complex. 
Windows-1252, CP1252, MSCP1252, and WINDOWS-1252 are not suitable for iexplore.exe.

So we should not change it unless problems don't happen.

See comments of files and directories below: 
https://git.savannah.gnu.org/gitweb/?p=libiconv.git;a=tree;f=lib,
https://git.savannah.gnu.org/gitweb/?p=libiconv.git;a=tree;f=libcharset/lib,
https://git.savannah.gnu.org/gitweb/?p=libiconv.git;a=blob;f=libcharset/lib/localcharset.c.

Traditional charsets are very complex.
---
wantfix

File: xsl/gnc-custom-chm.xsl (hunk 2 above)
Comment (usamimikamen-sk) [reply xsl/gnc-custom-chm.xsl:39]: wantfix

File: xsl/gnc-custom-chm.xsl (hunk 1 above)
Comment (fellen) [reply xsl/gnc-custom-chm.xsl:31]: We have a problem:
ISO-8859-1 has no `€` symbol, which is the main currency symbol in our docs in de, it, pt. It is also used in C.
That was the main reason to replace ISO-8859-1 by ISO-8859-15.

Comment (usamimikamen-sk) [reply xsl/gnc-custom-chm.xsl:31]: Ideally, you're right.
In practice, you haven't checked the output of the file.

Euro sign is coded by xsltproc to &#8364; in ISO-8859-1. And iexplore.exe decode and render it as Euro sign correctly. Therefore no problem will happen.

Moreover, HORIZONTAL_ELLIPSIS &#8230; is not rendered on TOC of htmlhelp.exe.

I think it isn't very sensible to stick to standards for the legacy CHM files. Stick to their outputs. Of course, it is NOT ideal. 

de-guide-ch03

Comment (usamimikamen-sk) [reply xsl/gnc-custom-chm.xsl:31]: de-guide-ch07

Comment (usamimikamen-sk) [reply xsl/gnc-custom-chm.xsl:31]: Another option is to stop using CHM. If we use HTML files, you will not bother to character sets. We can directly use UTF-8.

Comment (fellen) [reply xsl/gnc-custom-chm.xsl:31]: 7 byte instead of one. ;-)
I do not understand why you insists on using the obsolete ISO-8859-1 and spent days on research instead of changing it simply to the more recent ISO-8859-15.

Usually specific help browsers have some benefits over normal web browsers like extended search  facilities. As I am no Windows user, I have no experience and no strong opinion. As the currently used browser is no longer supported by MS, what are they offering instead?

Perhaps you should collect the opinion of Windows users over the mailing list.

Comment (usamimikamen-sk) [reply xsl/gnc-custom-chm.xsl:31]: From my point of view:
* ISO-8859-1 is the default encoding of xsltproc. And no one noticed problems over 10 years. Actually, no problem happens except Japanese and Russian.
* You requested me to change the encoding to UTF-8. I answered I can't expect what will happen in the TOC pane.
* You requested me to use Windows-1252. I answered it is not suitable for HTML. In fact, both documents and TOC are formatted with HTML internally for CHM.
* You requested me to use ISO-8859-15 because the Euro sign is not defined in ISO-8859-1. But the Euro sign is rendered in the HTML pane correctly. And I found you haven't either viewed or tested the output of CHM.

I beg you to view, test, and compare the files before requesting me to change the settings.
Your requests are not fit the reality of CHM.

>Usually specific help browsers have some benefits over normal web browsers like extended search facilities. As I am no Windows user, I have no experience and no strong opinion. As the currently used browser is no longer supported by MS, what are they offering instead?

About Windows, my opinion is the same as Microsoft. 
If you install gnucash-docs on the localhost, Windows index search will be used.
If you put gnucash-docs on the internet, search engines like google will be used.
Microsoft lead us to follow the guideline and they plan to remove the local help browser such as htmlhelp.exe and CHM, completely.

Comment (usamimikamen-sk) [reply xsl/gnc-custom-chm.xsl:31]: >About Windows, my opinion is the same as Microsoft.
>If you install gnucash-docs on the localhost, Windows index search will be used.
> If you put gnucash-docs on the internet, search engines like google will be used.

I think these MDN pages are our goal.
https://developer.mozilla.org/ja/docs/Learn/Getting_started_with_the_web/CSS_basics

File: cmake/AddGncDocTargets.cmake [hunk 3]
```diff
@@ -1,47 +1,68 @@
...
+    # can be replaced with list(TRANSFORM <list> <ACTION> [...])
+    file(GLOB_RECURSE figures_dist
+        RELATIVE ${CMAKE_CURRENT_SOURCE_DIR}
+        figures/*.png figures/*.svg)
+
+    set(dtd_files "${CMAKE_SOURCE_DIR}/docbook/gnc-docbookx.dtd"
+                  "${CMAKE_SOURCE_DIR}/docbook/gnc-locale-C.dtd"
+                  "${CMAKE_SOURCE_DIR}/docbook/gnc-locale-${lang}.dtd")
+    list(REMOVE_DUPLICATES dtd_files)
+    list(APPEND source_files ${dtd_files})
```
Comment (gjanssens) [inline cmake/AddGncDocTargets.cmake:31]: Variable source_files is not used at this level and reset in the functions we call later, so it can be removed here.

File: cmake/AddPdfTarget.cmake [hunk 4]
```diff
@@ -1,23 +1,46 @@
...
-    # * all others use A4.
-    set (XSLTFLAGS_FO "--stringparam paper.type A4")
-    if (lang STREQUAL "C")
-        # For the fallback language determine paper format depending on locale
-        # Only US or C will be set to letter. All others use A4.
-        set (ENV_LANG $ENV{LANG})
-        if (ENV_LANG AND ENV_LANG MATCHES ".*_us.*|C") # Replacing ENV_LANG here with if ($ENV{LANG}) won't work.
-            set (XSLTFLAGS_FO "--stringparam paper.type letter")
-        endif()
-    endif()
```
Comment (gjanssens) [inline cmake/AddPdfTarget.cmake:20]: Removing the page format resolution is a different topic from adding argument dtd_files. It should happen in a different commit. Ideally the commit that proposes another solution.

**Review by gjanssens — CHANGES_REQUESTED** [review]
My feedback so far pertains only to the first commit "Added an argument dtd_files to add_${fmt}_target

The commit actually handles four topics as illustrated by the full commit message. And this also illustrates why that's not a good idea.

1. It's not clear from  the long list of changes which dependency got fixed.
2. The main topic (adding a dtd_files argument) only requires two minor corrections for it  to be merged, but due to the wider questions around the build directory transition it is blocked.
3. It will make discussions on different topic more complicated as the changes are in one commit.

So please split this commit even further in the separate topics you mention in your commit message
1. add the dtd_files argument (including adding the comment sections to all the functions)
2. the dependency fix (unless you refer  to removing the docbook file as a direct dependency)
3. build directory changes which still needs a good motivation to actually implement.

**Comment by gjanssens** [comment]
The commit that merges MakeChm.cmake with AddChmTarget.cmake needs more testing. As mentioned in an earlier PR that's now closed, gnucash at the moment requires the hhmap file which is no longer built in your commit. I suggest to move this commit  to its own PR as it will surely require extensive testing and potentially synchronous changes in gnucash.

File: cmake/AddEpubTarget.cmake [hunk 5]
```diff
@@ -48,11 +47,35 @@ function (add_epub_target docname lang entities figures dtd_files)
...
+        COMMAND cp -f "${CMAKE_SOURCE_DIR}/xsl/icons/*" "${BUILD_DIR}/OEBPS/images"
+        COMMAND touch "${CMAKE_CURRENT_BINARY_DIR}/${fmt}-gnucashicon-trigger"
+        DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/${fmt}-xslticon-trigger"
+                "${gnucash_icon_files}")
+
 
     add_custom_command(
         OUTPUT "${BUILD_DIR}/${epubfile}"
-        COMMAND rm -fr "${EPUB_TMPDIR}"
-        COMMAND mkdir "${EPUB_TMPDIR}"
```
Comment (gjanssens) [inline cmake/AddEpubTarget.cmake:12]: (Nitpick) Not removing the build directory means your final epub document may contain html files and or images that are not used because we zip up the content of that directory unconditionally. This can happen if:
* the xsl transform changes the chunking (eg sections get renamed)
* images are removed from the source

That's the reason also the command depends on the figure source files rather than the figtrigger.

It's a minor issue, which I think I'll let pass. Builders can avoid this by building in a clean build directory.

File: cmake/AddGHelpTarget.cmake [hunk 6]
```diff
@@ -25,20 +25,11 @@ function (add_ghelp_target docname lang entities figures dtd_files)
...
-
     set(source_files "")
     foreach(xml_file ${entities} ${docname}.xml)
         list(APPEND source_files "${CMAKE_CURRENT_SOURCE_DIR}/${xml_file}")
     endforeach()
 
-    set(dtd_files "${CMAKE_SOURCE_DIR}/docbook/gnc-docbookx.dtd"
-                  "${CMAKE_SOURCE_DIR}/docbook/gnc-locale-C.dtd"
-                  "${CMAKE_SOURCE_DIR}/docbook/gnc-locale-${lang}.dtd")
-    list(REMOVE_DUPLICATES dtd_files)
```
Comment (gjanssens) [inline cmake/AddGHelpTarget.cmake:25]: These should have been removed in an earlier commit.

File: cmake/AddGHelpTarget.cmake [hunk 7]
```diff
@@ -25,20 +25,11 @@ function (add_ghelp_target docname lang entities figures dtd_files)
     # GnuCash-specific xsl files
     file(GLOB gnucash_icon_files "${CMAKE_SOURCE_DIR}/xsl/icons/*")
 
-    set(BUILD_DIR "${DATADIR_BUILD}/gnome/help/${docname}/${lang}")
-
```
Comment (gjanssens) [inline cmake/AddGHelpTarget.cmake:16]: That should have been done in an earlier commit.

File: cmake/AddGHelpTarget.cmake [hunk 8]
```diff
@@ -48,29 +39,47 @@ function (add_ghelp_target docname lang entities figures dtd_files)
...
-        OUTPUT ${dest_files}
-        COMMAND ${CMAKE_COMMAND} -E copy ${source_files} "${BUILD_DIR}"
-        DEPENDS ${entities} "${docname}.xml" ${dtd_files}
-                "${CMAKE_CURRENT_BINARY_DIR}/${fmt}-preparedir-trigger"
-        WORKING_DIRECTORY "${BUILD_DIR}")
+        OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/${fmt}-dtd-trigger"
+        COMMAND ${CMAKE_COMMAND} -E copy ${dtd_files} "${BUILD_DIR}"
+        COMMAND touch "${CMAKE_CURRENT_BINARY_DIR}/${fmt}-dtd-trigger"
+        DEPENDS "${dtd_files}"
+                "${CMAKE_CURRENT_BINARY_DIR}/${fmt}-preparedir-trigger")
```
Comment (gjanssens) [inline cmake/AddGHelpTarget.cmake:42]: Don't use the detour of  xyz-trigger to simply copy files in place. Use  ` configure_file` instead.  More importantly though the extra separation of docbook files from dest_files doesn't fit in this commit. For starters I don't really see what you gain by doing so and if there really is gain it's a separate topic so it should be done in a separate commit.

File: cmake/AddPdfTarget.cmake [hunk 9]
```diff
@@ -43,25 +40,52 @@ function (add_pdf_target docname lang entities figures dtd_files)
...
-                            -o "${CMAKE_CURRENT_BINARY_DIR}/${fofile}"
+                            -o "${BUILD_DIR}/${fofile}"
                             --stringparam fop1.extensions 1
                             "${CMAKE_SOURCE_DIR}/xsl/1.79.2/fo/docbook.xsl"
                             "${CMAKE_CURRENT_SOURCE_DIR}/${docname}.xml"
         DEPENDS ${entities} "${docname}.xml" ${dtd_files} ${xsl_files}
                 "${CMAKE_CURRENT_BINARY_DIR}/${fmt}-preparedir-trigger")
 
-    configure_file("${FOP_XCONF}" "${CMAKE_CURRENT_BINARY_DIR}/fop.xconf")
-
```
Comment (gjanssens) [inline cmake/AddPdfTarget.cmake:32]: Should have been done in an earlier commit.

File: guide/ja/fop.xconf.in [hunk 10]
```diff
@@ -2,7 +2,7 @@
 <fop version="1.0">
   <!-- BASEDIR_JA can be replaced with CMAKE_CURRENT_SOURCE_DIR
        once we completely drop the autotools build system -->
-  <base>@BASEDIR_JA@</base>
+  <base>${BUILD_DIR}</base>
```
Comment (gjanssens) [inline guide/ja/fop.xconf.in:3]: Remove the now obsolete comment as well.

File: guide/ru/fop.xconf.in [hunk 11]
```diff
@@ -2,7 +2,7 @@
 <fop version="1.0">
   <!-- BASEDIR_RU can be replaced with CMAKE_CURRENT_SOURCE_DIR
        once we completely drop the autotools build system -->
-  <base>@BASEDIR_RU@</base>
```
Comment (gjanssens) [inline guide/ru/fop.xconf.in:5]: Remove the now obsolete comment as well.

**Review by gjanssens — CHANGES_REQUESTED** [review]
Review of commit "Moved OASIS xslt icons to xsl/images".

Lots of small cleanups, and rework of how images and figures are copied.

**Comment by gjanssens** [comment]
On a more general note, I'd like you to drop the `${fmt}` variable from the  various add_<fmt>_target functions. It's not helpful because
1. Each time ${fmt} is used we need a mental jump to translate it to the proper format value.
2. That would be ok if the value was relative to some other parameter, but it's not. It's a fixed value that's not dependent on any build environment parameters (unlike BUILD_DIR which is different in each build environment because of the base path)
3. the values themselves are shorter than the variable name (3-4 characters vs 6 characters) so it's actually more work to type the variable than it is to type to value.

**Comment by gjanssens** [comment]
> Review of commit "Moved OASIS xslt icons to xsl/images".
> 
> Lots of small cleanups, and rework of how images and figures are copied.

One more thing: you removed the stylesheets directory but it's still referenced in  extra_files in the top-level CMakeLists.txt. Please remove it there as well.

File: CMakeLists.txt [hunk 12]
```diff
@@ -183,12 +161,10 @@ add_subdirectory (manual)
...
 file(GLOB_RECURSE extrafiles
     RELATIVE ${CMAKE_CURRENT_SOURCE_DIR}
-    cmake/* fonts/* stylesheet/* xsl/* docbook/*)
+    cmake/* fonts/* xsl/* docbook/*)
 add_to_dist(${extrafiles})
 
-configure_file(COPYING ${DATADIR_BUILD}/${PROJECT_NAME}/COPYING COPYONLY)
-configure_file(COPYING-DOCS ${DATADIR_BUILD}/${PROJECT_NAME}/COPYING-DOCS COPYONLY)
-install(FILES COPYING COPYING-DOCS DESTINATION ${CMAKE_INSTALL_DATADIR}/${PROJECT_NAME})
+install(FILES COPYING COPYING-DOCS DESTINATION "share/doc/${PROJECT_NAME}")
```
Comment (gjanssens) [inline CMakeLists.txt:167]: Your choice of "doc" as subdirectory is nice, though this change will require coordination with @derekatkins to update the nightly build scripts. Please do it in a separate commit to unblock the rest of the changes.

On the other hand I'm uncomfortable with hardcoding  'share' rather than using  CMAKE_INSTALL_DATADIR. I understand you want to drop module GNUInstallDirs to avoid having to set a (programming) language, but this limits configurability and may cause certain distros to no longer be able to install properly.

If you don't want GNUInstallDirs, you should define a cache variable intitially set to "share" in the top level CMakeLists.txt  file and use that everywhere (including in the build steps that output in (a subdirectory) of builddir/share). That allows packagers to change the destination path as needed.

File: cmake/AddChmTarget.cmake [hunk 13]
```diff
@@ -42,7 +43,6 @@ function (add_chm_target docname lang entities figures dtd_files)
         WORKING_DIRECTORY "${BUILD_DIR}"
         DEPENDS "${BUILD_DIR}/htmlhelp.hhp"
                 "${CMAKE_CURRENT_BINARY_DIR}/${fmt}-fig-trigger"
-                "${CMAKE_CURRENT_BINARY_DIR}/${fmt}-xslticon-trigger"
```
Comment (gjanssens) [inline cmake/AddChmTarget.cmake]: Why do you drop this dependency ? Is htmlhelp.hpp not dependent on xslt icons but it is on figures ?

File: cmake/AddChmTarget.cmake [hunk 14]
```diff
@@ -52,7 +52,6 @@ function (add_chm_target docname lang entities figures dtd_files)
         OUTPUT "${BUILD_DIR}/htmlhelp.hhp"
         COMMAND ${XSLTPROC} ${XSLTPROCFLAGS} ${XSLTPROCFLAGS_CHM}
                 -o "${BUILD_DIR}/"
-                --path "${CMAKE_SOURCE_DIR}/docbook"
```
Comment (gjanssens) [inline cmake/AddChmTarget.cmake]: If this is no longer needed here, it should probably have been removed the moment the docbook changes were made.

File: cmake/AddChmTarget.cmake [hunk 15]
```diff
@@ -88,14 +87,14 @@ function (add_chm_target docname lang entities figures dtd_files)
 
 
 
-    add_custom_target("${lang}-${docname}-chm"
+    add_custom_target("${lang}-${docname}-${fmt}"
```
Comment (gjanssens) [inline cmake/AddChmTarget.cmake:90]: I'm in favour of keeping the hardcoded format names for reasones already mentioned in an earlier comment.

File: cmake/AddEpubTarget.cmake [hunk 16]
```diff
@@ -36,15 +36,13 @@ function (add_epub_target docname lang entities figures dtd_files)
     file(GLOB xsl_files "${CMAKE_SOURCE_DIR}/xsl/*.xsl")
     file(GLOB gnucash_icon_files "${CMAKE_SOURCE_DIR}/xsl/icons/*")
 
-    set(epubfile "${docname}.epub")
-    set(EPUB_TMPDIR "${BUILD_DIR}")
-
```
Comment (gjanssens) [inline cmake/AddEpubTarget.cmake]: That should go in the commit that redefines EPUB_TMPDIR.

File: cmake/AddGHelpTarget.cmake [hunk 17]
```diff
@@ -65,28 +66,26 @@ function (add_ghelp_target docname lang entities figures dtd_files)
...
                 "${CMAKE_CURRENT_BINARY_DIR}/${fmt}-dtd-trigger"
                 "${CMAKE_CURRENT_BINARY_DIR}/${fmt}-gnucashicon-trigger"
                 "${CMAKE_CURRENT_BINARY_DIR}/${fmt}-fig-trigger")
 
-    add_dependencies(${docname}-ghelp "${lang}-${docname}-ghelp")
+    add_dependencies(${docname}-${fmt} "${lang}-${docname}-${fmt}")
+
+    install(DIRECTORY ${OUTPUT_DIR}
+        DESTINATION "share/doc/${lang}"
+        COMPONENT "${fmt}")
```
Comment (gjanssens) [inline cmake/AddGHelpTarget.cmake:80]: Basing your install on whatever is in `${OUTPUT_DIR}` makes it susceptible to installing leftover files from previous builds. I'm in two minds about whether that's an important enough issue to not do it.

Anyway the use of ${fmt} is  still not helpful.

File: cmake/AddHtmlTarget.cmake [hunk 18]
```diff
@@ -79,9 +79,11 @@ function (add_html_target docname lang entities figures dtd_files)
                 "${CMAKE_CURRENT_BINARY_DIR}/${fmt}-fig-trigger"
                 "${CMAKE_CURRENT_BINARY_DIR}/${fmt}-gnucashicon-trigger")
 
-    add_dependencies(${docname}-html "${lang}-${docname}-html")
+    add_dependencies(${docname}-${fmt} "${lang}-${docname}-${fmt}")
+
+    install(DIRECTORY ${OUTPUT_DIR}
+        DESTINATION "share/doc/${lang}"
+        OPTIONAL
+        COMPONENT "${fmt}")
```
Comment (gjanssens) [inline cmake/AddHtmlTarget.cmake:85]: Adding optional is a good catch.

File: cmake/AddPdfTarget.cmake [hunk 19]
```diff
@@ -29,16 +29,16 @@ function (add_pdf_target docname lang entities figures dtd_files)
...
 
-    set(pdffile "${docname}.pdf")
-
 
     # Prepare ${BUILD_DIR}
     add_custom_command(
         OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/${fmt}-preparedir-trigger"
         COMMAND ${CMAKE_COMMAND} -E make_directory "${BUILD_DIR}/figures"
         COMMAND ${CMAKE_COMMAND} -E make_directory "${BUILD_DIR}/images/callouts"
+        COMMAND ${CMAKE_COMMAND} -E make_directory "${OUTPUT_DIR}"
```
Comment (gjanssens) [inline cmake/AddPdfTarget.cmake:38]: Like all the other directories, please use cmake's `file(ADD_DIRECTORY ...)` rather than a custom command. As explained before we don't need to track existence of directories as they don't play in dependency resolution.

File: cmake/AddPdfTarget.cmake [hunk 20]
```diff
@@ -86,10 +87,14 @@ function (add_pdf_target docname lang entities figures dtd_files)
...
 
-    add_custom_target("${lang}-${docname}-pdf"
-        DEPENDS "${BUILD_DIR}/${pdffile}")
+    add_dependencies(${docname}-${fmt} "${lang}-${docname}-${fmt}")
 
-    add_dependencies(${docname}-pdf "${lang}-${docname}-pdf")
+    install(FILES "${OUTPUT_DIR}/${outfile}"
+        DESTINATION "share/doc/${lang}"
+        OPTIONAL
+        COMPONENT "${fmt}")
```
Comment (gjanssens) [inline cmake/AddPdfTarget.cmake:98]: As explained on an earlier PR we don't need install targets for pdf, epub and mobi. If you really want to add those, that should  go in a separate commit.

I hope by now you get the picture: one topic per commit. It's incredibly confusing trying to figure out why a certain change was made if a single commit mixes different goals.

File: guide/ja/CMakeLists.txt [hunk 21]
```diff
@@ -22,11 +22,6 @@ set (entities
         appendixd.xml)
 
 set (FOP_XCONF "${CMAKE_CURRENT_SOURCE_DIR}/fop.xconf.in")
-# Compatibility hack for autotools
-# When autotools gets dropped this line can be removed
-# and CMAKE_CURRENT_SOURCE_DIR can be used directly in
-# fop.xconf.in
-set(BASEDIR_JA "${CMAKE_CURRENT_SOURCE_DIR}")
```
Comment (gjanssens) [inline guide/ja/CMakeLists.txt:29]: Should be removed in the commit that updates the variable in fop.xconf.in.

File: guide/ja/fop.xconf.in [hunk 22]
```diff
@@ -1,7 +1,5 @@
 <?xml version="1.0"?>
 <fop version="1.0">
-  <!-- BASEDIR_JA can be replaced with CMAKE_CURRENT_SOURCE_DIR
-       once we completely drop the autotools build system -->
```
Comment (gjanssens) [inline guide/ja/fop.xconf.in:4]: Move it  to the commit that actually changes this parameter.

File: guide/ru/CMakeLists.txt [hunk 23]
```diff
@@ -24,11 +24,6 @@ set (entities
         appendixd.xml)
 
 set (FOP_XCONF "${CMAKE_CURRENT_SOURCE_DIR}/fop.xconf.in")
-# Compatibility hack for autotools
-# When autotools gets dropped this line can be removed
-# and CMAKE_CURRENT_SOURCE_DIR can be used directly in
-# fop.xconf.in
-set(BASEDIR_RU "${CMAKE_CURRENT_SOURCE_DIR}")
```
Comment (gjanssens) [inline guide/ru/CMakeLists.txt:31]: Same here. Move it to the commit that changes the russian fop.xconf.in.

**Review by gjanssens — CHANGES_REQUESTED** [review]
Review of "Changed the layout of ${OUTPUT_DIR}.

There are still plenty of small changes  that don't belong in this commit. Several should have been done already in earlier commits, others should go into their own separate commit (like adding install targets).

File: cmake/AddPdfTarget.cmake [hunk 24]
```diff
@@ -43,7 +43,7 @@ function (add_pdf_target docname lang entities figures dtd_files)
         OUTPUT "${BUILD_DIR}/${fofile}"
         COMMAND ${XSLTPROC} ${XSLTPROCFLAGS} ${XSLTPROCFLAGS_FO}
                             -o "${BUILD_DIR}/${fofile}"
-                            --stringparam fop1.extensions 1
+                            --stringparam gnc.lang ${lang}
```
Comment (gjanssens) [inline cmake/AddPdfTarget.cmake:46]: Ok, though the  cmake bit that defined  the papersize in the past should only be removed as part of this commit.

File: xsl/gnc-custom-chm.xsl (hunk 1 above)
Comment (gjanssens) [reply xsl/gnc-custom-chm.xsl:31]: > Windows code pages are not compliant with ISO codes like ISO-8859-*. Moreover, the multibyte charset is more complex. Windows-1252, CP1252, MSCP1252, and WINDOWS-1252 are not suitable for iexplore.exe because they are not defined by IANA.

Why are you referring to iexplore here ? I thought chm was using a custom help viewer. The change only applies to the chm format which is as you write yourself old and not utf-8 capable (which IE is by the way) so it seems to me iexplore  is irrelevant here ?

Other than that I have no idea whether there's a problem here or not and how to potentially fix it.

Comment (gjanssens) [reply xsl/gnc-custom-chm.xsl:31]: Changing to html is probably a reasonable choice, though that needs coordinated changes in gnucash-on-windows and  gnucash. And testing to check if it all works as we want it to.
